src/backend/api/meals.js

```diff
@@ -2,13 +2,188 @@ const express = require("express");
+
+    //Returns all meals where the date for meal_time is before the given date.(api/meals?dateBefore=2024-01-01)
+    if (dateBefore) {
+      console.log({ dateBefore, formatted: new Date(dateBefore) }); //meal_time is in datetime format in DB
+      const result = await knex
+        .select("*")
+        .from("meal")
+        .where("meal_time", "<", new Date(dateBefore));
+
+      response.data = result;
+      response.status = 200;
+      response.message = "Meals with date before the given date";
+    }
+
+    //Returns the given number of meals (api/meals?limit=7)
+    if (limit) {
+      const result = await knex
+        .select("*")
+        .from("meal")
+        .orderBy("id", "asc")
+        .limit(+limit);
+
+      response.data = result;
+      response.status = 200;
+      response.message = `Here are the first ${limit} meals`;
+    }
+
+    //Returns all meals sorted by the given key. Allows when, max_reservations and price as keys.
+    //Default sorting order is asc(ending). (api/meals?sortKey=price)
+    if (["meal_time", "max_reservations", "price"].includes(sortKey)) {
+      const result = await knex
+        .select("*")
+        .from("meal")
+        .orderBy(sortKey, sortDir === "desc" ? sortDir : "asc");
+
+      response.data = result;
+      response.status = 200;
+      response.message = "Meals sorted by the given sortKey";
+    }
+
+    //Returns all meals sorted in the given direction.
+    //Only works combined with the sortKey and allows asc or desc. (api/meals?sortKey=price&sortDir=desc)
+    if (["meal_time", "max_reservations", "price"].includes(sortKey)) {
+      if (sortDir === "asc" || sortDir === "desc") {
+        const result = await knex
+          .select("*")
+          .from("meal")
+          .orderBy(sortKey, sortDir);
+
+        response.data = result;
+        response.status = 200;
+        response.message =
+          "Meals sorted by the given sort key and sort direction";
+      }
+    }
+
+    // Returns all meals if no filter is given
+    if (Object.keys(req.query).length === 0) {
+      const result = await knex.select("*").from("meal");
+
+      response.data = result;
+      response.status = 200;
+      response.message = "Here are all meals, as no filter key has been given";
+    }
+
+    console.log(response);
+
+    res
+      .status(response.status)
+      .json({ data: response.data, message: response.message });
+  } catch (error) {
+    console.error("Server Error", error);
+    res.status(500).json({ error: "Internal server error" });
+  }
+});
+
+//Returns all reviews for a specific meal. (/api/meals/:mealid/reviews)
+router.get("/:mealid/reviews", async (req, res) => {
+  const mealid = +req.params.mealid;
+  const response = {};
+  try {
+    if (mealid) {
+      const result = await knex
+        .select("meal.title", "review.description")
+        .from("meal")
+        .join("review", "meal.id", "=", "review.meal_id")
+        .where("meal.id", "=", mealid);
```

Comment:
Is there a need to perform the join here? You already have the ID of the meal, which is the actual value of the `meal_id` foreign key of the `review` table. You can just simply select from the `review` table. You would only need to perform this join if you had the name of the meal, or something that you had to look up in the `meal` table. 🧠